design-documents/graph-ql/coverage/negotiableQuotes.graphqls

```diff
@@ -0,0 +1,147 @@
+type Mutation {
+    createNegotiableQuote(
+        cart_id: ID! @doc(description: "Cart ID")
+        quote_name: String @doc(description: "Quote name")
+        comment: String @doc(description: "Comment")
+        files: [FileInput!] @doc(description: "Attached files")
+    ): RequestNegotiableQuoteOutput! @doc(description:"Generates negotiable quote request")
+
+    setShippingMethodsOnNegotiableQuote(
+        cart_id: ID! @doc(description: "Cart ID")
+        shipping_methods: [String!]! @doc(description: "Input for shipping method")
+    ): SetShippingMethodsOnNegotiableQuoteOutput @doc(description:"Set negotiable quote shipping method")
+}
+
+type Customer {
+    negotiable_quote (
+        id: ID!): NegotiableQuote @doc(description: "Get negotiable quote of a customer")
+
+    negotiable_quotes(
+        page_size: Int = 20,
+        current_page: Int = 1,
+        filter: NegotiableQuoteFilterInput
+    ): [NegotiableQuote] @doc(description: "Get negotiable quotes of a customer")
+
+    estimateShippingCostForNegotiableQuote(
+        address: NegotiableQuoteShippingAddressInput!
+    ): [ShippingMethodsOutput]! @doc(description:"Estimate shipping costs for a negotiable quote")
+
+    estimateShippingCostByAddressIdForNegotiableQuote(
+        address_id: String!
+    ): [ShippingMethodsOutput]! @doc(description:"Estimate shipping costs for a negotiable quote by Address ID")
+}
+
+type NegotiableQuote {
+    quote_id: ID!  @doc(description: "Negotiable quote ID")
+    items: [NegotiableQuoteItem] @doc(description: "Negotiable quote item")
+    attachements: [AttachmentContent] @doc(description: "Negotiable quote Attachements")
+    comments: [NegotiableQuoteComment] @doc(description: "Negotiable quote comments")
+    totals: TotalsOutput! @doc(description: "Negotiable quote totals output")
+    payment_info: NegotiableQuotePaymentInfo! @doc(description: "Negotiable quote payment info.")
+    available_shipping_methods: [ShippingMethodsOutput]! @doc(description: "Available shipping methods")
+}
+
+input NegotiableQuoteFilterInput {
+    ids: FilterEqualTypeInput @doc(description: "Filter Customer Negotiable quotes with an negotiable quote ID or list of negotiable quote IDs")
+    name: FilterMatchTypeInput @doc(description: "Filter by display name of the negotiable quote")
+}
+
+type NegotiableQuoteItem {
+    id: ID!
+    quantity: Float!
+    product: ProductInterface!
+}
+
+type AttachmentContent @doc(description: "Negotiable quote attachment file") {
+    base64_encoded_data: String!
+    mime_type: String!
+    name: String!
+}
+
+type NegotiableQuoteComment {
+    entity_id: ID!
+    parent_id: ID!
+    creator_type: Int
+    is_decline: Int
+    is_draft: Int
+    creator_id: ID!
+    comment: String
+    created_at: String
+    attachments: [NegotiableQuoteCommentAttachments]
```

Comment:
Should be an array of links, like product images and downloadable products. If more metadata is necessary custom type should be introduced, but still must not expose IDs and file path  like in current version of `NegotiableQuoteCommentAttachments`
```suggestion
    attachments: [String]
```